File: 210/level1/음양더하기.java
// 20220912

package level1;

class Main {
    public static void main(String[] args) throws Exception {
        int[] absolutes = {4,7,12};
        boolean[] signs = {true,false,true};

        Solution sol = new Solution();
        System.out.println("result : " + sol.solution(absolutes, signs));
    }
}

class Solution {
    public int solution(int[] absolutes, boolean[] signs) {
        int answer = 0;
        int length = absolutes.length;

        for (int i = 0; i < length; i++) {
            answer = signs[i] ? answer + absolutes[i] : answer - absolutes[i];
        }

        return answer;
    }
}

File: JunJangE/level1/음양 더하기.kt
class Solution {
    fun solution(absolutes: IntArray, signs: BooleanArray): Int {
        var answer: Int = 0

        absolutes.forEachIndexed { index, i ->

            if (signs[index]){
                answer += i
            }else{
                answer -= i
            }
        }
        return answer
    }
}
